app/controllers/admin/projects_controller.rb

```diff
@@ -19,8 +19,14 @@ def steps
   def timeline
     @project = Project.find(params[:id])
     authorize @project, :show?
-    prep_timeline(@project)
-    render partial: "admin/timeline/table", locals: {project: @project}
+    if request.xhr?
+      prep_timeline(@project)
+      render partial: "admin/timeline/table", locals: {project: @project}
+    elsif @project.is_a? Loan
+      redirect_to admin_loan_tab_path(@project, tab: 'timeline')
+    elsif @project.is_a? BasicProject
+      redirect_to admin_basic_project_tab_path(@project, tab: 'timeline')
```

Comment:
`is_a?` is often a code smell (see Metz book, duck typing is usually better) but here I don't see a way to do that without coupling the route logic to the Loan/BasicProject models...

Although you could have a helper method here called `project_tab_path` that does something like `send("admin_#{@project.model_name.singular)_tab_path", ...)` behind the scenes. I think that would be preferable.